tests/Integration/Behaviour/Features/Context/Domain/OrderFeatureContext.php

```diff
@@ -183,17 +184,21 @@ public function addProductsToOrderWithNewInvoiceAndTheFollowingDetails(string $o
 
         $productName = $data['name'];
         $product = $this->getProductByName($productName);
-
         $productId = $product->getProductId();
-        if (isset($data['combination'])) {
-            $combinationId = $this->getProductCombinationId($product, $data['combination']);
-        } else {
-            $combinationId = 0;
-        }
+        $productCore = new Product((int) $productId);
+        $ecotax = new Number($productCore->ecotax);
+
+        $combinationId = isset($data['combination']) ? $this->getProductCombinationId($product, $data['combination']) : 0;
 
         if (empty($data['price_tax_incl'])) {
-            $taxCalculator = $this->getProductTaxCalculator((int) $orderId, (int) $productId);
-            $data['price_tax_incl'] = !empty($taxCalculator) ? (string) $taxCalculator->addTaxes($data['price']) : $data['price'];
+            $taxCalculator = $this->getProductTaxCalculator((int) $orderId, $productId);
```

Comment:
I'm not sure I understand this.. you perform the same computing as in the handler, basically, so of course it will match And it's a very suspicious computing as I commented earlier 😅

Don't get me wrong I understand why you needed to add this, but it feels so illogical that I feel the need to ask for blocking specifications here.. It feels like we are forcing a square in a circle just to please the legacy behaviour

Reply:
Fixed